test/filesystem.cc

```diff
@@ -135,6 +137,139 @@ TEST(Filesystem, extension) {
 TEST(Filesystem, file_size) {
 }
 
+TEST(Filesystem, concurrent_folder_create_delete) {
```

Comment:
Great test! Very clean and readable. Thank you!

BTW, do we need a test-level cleanup function to remove any leftover artifacts of the test? Or would rather have the directories around in case its useful for debugging test failures?

Reply:
Thanks and good thought.  However, as written, the test (concurrently) creates the nested subfolder (starting at "test/a"), populates it, then concurrently removes the "test/a" folder.  If the test is working, there shouldn't be any artifacts leftover.